src/twisted/mail/newsfragments/9180.bugfix

```diff
@@ -0,0 +1 @@
+Sending a list of recipients with twisted.smtp.SenderFactory has been fixed.  This fixes a problem found when running buildbot.
```

Comment:
@exarkun thanks for the additional feedback.  @tardyp reviewed the request.

I may not be as talented as some of the other Twisted devs, but I do take my commit privileges responsibly, to the best of my ability.